src/main/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserController.java

```diff
@@ -65,21 +67,38 @@
             value = "/users",
             produces = MediaType.APPLICATION_JSON_UTF8_VALUE
     )
-    @Secured("pui-user-manager")
+    @Secured({"pui-finance-manager", "pui-user-manager", "pui-organisation-manager", "pui-case-manager", "caseworker-divorce-financialremedy", "caseworker-divorce-financialremedy-solicitor", "caseworker-divorce-solicitor", "caseworker-divorce", "caseworker"})
     public ResponseEntity<?> findUsersByOrganisation(@ApiParam(hidden = true) @OrgId String organisationIdentifier,
                                                      @ApiParam(name = "showDeleted", required = false) @RequestParam(value = "showDeleted", required = false) String showDeleted,
-                                                     @ApiParam(name = "email", required = false) @RequestParam(value = "email", required = false) String email) {
+                                                     @ApiParam(name = "email", required = false) @RequestParam(value = "email", required = false) String email,
+                                                     @ApiParam(name = "status", required = false) @RequestParam(value = "status", required = false) String status) {
 
-        ResponseEntity<?> profUsersEntityResponse = null;
+        ResponseEntity<?> profUsersEntityResponse;
         log.info("ProfessionalExternalUserController::findUsersByOrganisation:" + organisationIdentifier);
-        profExtUsrReqValidator.validateRequest(organisationIdentifier, showDeleted, email);
+        profExtUsrReqValidator.validateRequest(organisationIdentifier, showDeleted, email, status);
+        ServiceAndUserDetails serviceAndUserDetails = (ServiceAndUserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        boolean isRolePuiUserManager = organisationIdentifierValidatorImpl.ifUserRoleExists(serviceAndUserDetails.getAuthorities(), "pui-user-manager");
 
         if (!StringUtils.isEmpty(email)) {
             log.info("email not empty");
-            profUsersEntityResponse = retrieveUserByEmail(email);
+            if (isRolePuiUserManager) {
+                profUsersEntityResponse = retrieveUserByEmail(email);
+            } else {
+                throw new InvalidRequest("Your role does not permit you to search for a user by email");
+            }
         } else {
-            log.info("showDeleted not empty");
-            profUsersEntityResponse = searchUsersByOrganisation(organisationIdentifier, showDeleted);
+
+            if (isRolePuiUserManager) {
+                log.info("Searching for users with Pui User Manager role");
+                profUsersEntityResponse = searchUsersByOrganisation(organisationIdentifier, showDeleted, true, status);
+
+            } else {
+                profExtUsrReqValidator.validateStatusIsActive(status);
+
+                log.info("Searching for users with NON Pui User Manager role");
+                status = "Active";
```

Comment:
why we have hardcoded status value ? . Instead of use OrganisationStatus.ACTIVE.name().